### DISCUSSION TITLE
IDX40001: Issuer 'https://[Instance]/[TenantId]/v2.0' does not match any of the valid issuers provided for this application.

### DISCUSSION THREAD

**Discussion by johnW-ret** [AzureAD/microsoft-identity-web#2481] [Q&A]
I had an ASP.NET Core app hosted in Azure Container Apps that was working with an app registered in an Azure AD B2C instance. When diagnosing a different issue, I had to delete my instance from Azure and redeploy. Now, my app with my exact same configuration, source code, and domains, does not work.

I keep getting the above error after signing in with my sign up and sign in user flow from my application.

```
IDX40001: Issuer 'https://[Instance]/[TenantId]/v2.0' does not match any of the valid issuers provided for this application.
```

In reference to #168, I have tried disabling issuer validation at least temporarily yet to no avail.

```csharp
builder.Services.Configure<OpenIdConnectOptions>(OpenIdConnectDefaults.AuthenticationScheme, opt => opt.TokenValidationParameters.ValidateIssuer = false);
```

Because it was working prior, I can't imagine this is anything but user error, but after poring over the docs, I'm a bit lost on how to next diagnose.

---

Here is my AzureAD configuration:
```json
  "AzureAdB2C": {
    "Authority": "https://[Instance]/[Domain]/B2C_1_signupandsignin1/v2.0/",
    "CallbackPath": "/signin-oidc",
    "ClientId": "...",
    "Domain": "...",
    "EditProfilePolicyId": "B2C_1_editprofile",
    "Instance": "...",
    "ResetPasswordPolicyId": "b2c_1_reset",
    "SignedOutCallbackPath": "/signout/B2C_1_signupandsignin1",
    "SignUpSignInPolicyId": "B2C_1_signupandsignin1",
    "TenantId": "..."
  },
```

dotnet --info
```
.NET SDK:
 Version:   8.0.100-rc.1.23463.5
 Commit:    e7f4de8816
...
.NET runtimes installed:
  ...
  Microsoft.AspNetCore.App 8.0.0-rc.1.23421.29 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
```

csproj
```
		<PackageReference Include="Microsoft.Identity.Web" Version="2.13.4" />
		<PackageReference Include="Microsoft.Identity.Web.UI" Version="2.13.4" />
```

**Comment by g-fleming** [comment]
I am also seeing this after migrating Blazor server app fron .net 7 to .net 8. Azure AD B2C give this error:

IDX40001: Issuer: 'https://<azure_A2d_name>.b2clogin.com/12345e9a-25bd-4b99-b0cf-4ac6d8b8ac5b/v2.0/', does not match any of the valid issuers provided for this application.

It worked previously.

Edit:

Updated Identity.Web Nuget package today and issue was resolved.

**Comment by jmprieur** [comment]
Did you try 2.15.2 (release yesterday) ? This should solve this issue

**Comment by jaliyaudagedara** [comment]
@jmprieur 

I am experiencing this issue.

```
[21:56:07 INF] Failed to validate the token.
Microsoft.IdentityModel.Tokens.SecurityTokenInvalidIssuerException: IDX40001: Issuer: 'https://xxxxx.b2clogin.com/<TenantId>/v2.0/', does not match any of the valid issuers provided for this application.
   at Microsoft.IdentityModel.Validators.AadIssuerValidator.ValidateAsync(String issuer, SecurityToken securityToken, TokenValidationParameters validationParameters)
   at Microsoft.IdentityModel.Validators.AadIssuerValidator.Validate(String issuer, SecurityToken securityToken, TokenValidationParameters validationParameters)
   at Microsoft.IdentityModel.Tokens.Validators.ValidateIssuerAsync(String issuer, SecurityToken securityToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
   at Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler.ValidateTokenPayloadAsync(JsonWebToken jsonWebToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
   at Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler.ValidateJWSAsync(JsonWebToken jsonWebToken, TokenValidationParameters validationParameters, BaseConfiguration configuration)
```

I am on `.NET 8` and `Microsoft.Identity.Web: 2.15.5`

![image](https://github.com/AzureAD/microsoft-identity-web/assets/5653381/9c8b995a-b7a4-44d9-95ff-a55423e15c0e)

**Reply by brijesh-1111** [reply]
I was having something similar (again not sure of the same issue but the symptoms are similar) see my below post.

**Comment by brijesh-1111** [comment]
Same issue with Microsoft.Identity.Web  version 2.16.0

**Reply by brijesh-1111** [reply]
My bad. after my post, I came across https://github.com/AzureAD/microsoft-identity-web/issues/1249 where I found the issue was with my issuer. I was using my custom domain and from other side, the token was carrying b2c...

**Comment by massman07** [comment]
Has anyone figured out how to solve this? 

I am having same issue. Using 2.17.0
I also posted about it here: 

https://learn.microsoft.com/en-us/answers/questions/1574868/idx40001-issuer-does-not-match-any-of-the-valid-is

**Reply by jaliyaudagedara** [reply]
You might want to have a look at this:
https://github.com/dotnet/aspnetcore/issues/52283

**Reply by massman07** [reply]
Thanks for the suggestion. I went through the post and tried all the suggested solutions and I am still getting the same issue.

**Comment by jmprieur** [comment]
You need to do something extra. See https://github.com/AzureAD/microsoft-identity-web/wiki/Deploying-Web-apps-to-App-services-as-Linux-containers

**Reply by massman07** [reply]
I'm not sure what to look for in the link you suggested. I'm not using containers.
